server/src/controllers/mealControllers.js

```diff
@@ -1,86 +1,115 @@
-import meals from "../db/meals";
+import fs from "fs";
+import Meal from "../models/meals";
 
 class MealControllers {
-  //Get Meal for the Single day
-  static getMeal(req, res) {
-    return res.status(200).json({
-      status: res.statusCode,
-      data: meals
-    });
+  static async getMeal(req, res) {
+    try {
+      const meals = await Meal.findAll({
+        where: { catererId: 2 }
+      });
+      return res.status(200).json({
+        status: "success",
+        message: "Meals Retrieved",
+        data: meals
+      });
+    } catch (err) {
+      return res.status(500).json({
+        status: "error",
+        message: "Failed to Retrieve Meals"
+      });
+    }
   }
 
-  // Post/add a Meal Option
-  static postMeal(req, res) {
-    if (!req.body.name) {
-      return res.status(400).send({
-        status: res.statusCode,
-        message: "Meal name is required"
+  static async postMeal(req, res) {
+    try {
+      const { name, price } = req.body;
+      const { image } = req.files;
+      const imageUrl = `/src/images/${image.name}`;
+      const meal = await Meal.create({
+        name,
+        price,
+        imageUrl,
+        catererId: 2
       });
-    }
-    const newMeal = {
-      id: meals.length + 1,
-      name: req.body.name,
-      price: req.body.price
-    };
-    meals.push(newMeal);
-    return res.status(201).json({
-      status: res.statusCode,
-      data: [
-        {
-          id: newMeal.id,
-          name: newMeal.name,
-          price: newMeal.price
+      await image.mv(`.${imageUrl}`);
+      return res.status(201).json({
+        status: "success",
+        message: "Meal Option Added",
+        data: {
+          id: meal.id,
+          name: meal.name,
+          price: meal.price,
+          imageUrl: meal.imageUrl
         }
-      ]
-    });
+      });
+    } catch (err) {
+      return res.status(500).json({
+        status: "error",
+        message: err.message
+      });
+    }
   }
 
-  // put/Patch for updating the info of a meal
-  static updateMealName(req, res) {
-    const findMeal = meals.find(
-      meal => meal.id === parseInt(req.params.id, 10)
-    );
-    if (findMeal) {
-      delete findMeal.name;
-      findMeal.name = req.body.name;
-
+  static async updateMealName(req, res) {
```

Comment:
Whatever is going on in this controller should be checked again. I think it's overengineered